### PR TITLE
Fix intermittent test failure in Subproviders package

### PR DESCRIPTION
**Pull request by fabioberger**
## Description

Addresses: https://github.com/0xProject/0x-monorepo/issues/809

Solution: instead of sending a network request to a non-existent endpoint, we stub the `handleRequest` method for the faulty subprovider. 

## Testing instructions

Run the subproviders tests.

## Types of changes

*   [x] Bug fix (non-breaking change which fixes an issue)
*   [ ] New feature (non-breaking change which adds functionality)
*   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

*   [ ] Change requires a change to the documentation.
*   [ ] Added tests to cover my changes.
*   [ ] Added new entries to the relevant CHANGELOG.jsons.
*   [ ] Labeled this PR with the 'WIP' label if it is a work in progress.
*   [ ] Labeled this PR with the labels corresponding to the changed package.

### CODE REVIEW

File: packages/subproviders/test/unit/redundant_rpc_subprovider_test.ts [hunk 1]
```diff
@@ -41,6 +42,9 @@ describe('RedundantSubprovider', () => {
         const nonExistentSubprovider = new RpcSubprovider({
```
Comment (feuGeneA) [inline packages/subproviders/test/unit/redundant_rpc_subprovider_test.ts]: this should be deleted now, right?

File: packages/subproviders/test/unit/redundant_rpc_subprovider_test.ts [hunk 2]
```diff
@@ -41,6 +42,9 @@ describe('RedundantSubprovider', () => {
...
             rpcUrl: 'http://does-not-exist:3000',
         });
+        const handleRequestStub = Sinon.stub(nonExistentSubprovider, 'handleRequest').throws(
+            new Error('REQUEST_FAILED'),
```
Comment (feuGeneA) [inline packages/subproviders/test/unit/redundant_rpc_subprovider_test.ts]: this should be deleted now, right?

File: packages/subproviders/test/unit/redundant_rpc_subprovider_test.ts (hunk 1 above)
Comment (feuGeneA) [inline packages/subproviders/test/unit/redundant_rpc_subprovider_test.ts]: this should be deleted now, right?

File: packages/subproviders/test/unit/redundant_rpc_subprovider_test.ts [hunk 3]
```diff
@@ -41,6 +42,9 @@ describe('RedundantSubprovider', () => {
         const nonExistentSubprovider = new RpcSubprovider({
             rpcUrl: 'http://does-not-exist:3000',
```
Comment (feuGeneA) [inline packages/subproviders/test/unit/redundant_rpc_subprovider_test.ts]: this should be deleted now, right?

Comment (fabioberger) [reply packages/subproviders/test/unit/redundant_rpc_subprovider_test.ts]: No, we still need to pass in a value to `rpcUrl` at instantiation.

**Comment by fabioberger** [comment]
Is this passing consistently for you @feuGeneA?

**Pull request merged**